### ISSUE TITLE
pcustom recursive dereference

### ISSUE THREAD

**Issue by xmikasax** [hugsy/gef#549]
Can we call apply_structure_to_address recursively for address that pointer points to and not just remove LP_ part?

https://github.com/hugsy/gef/blob/master/gef.py#L4730

Right now it does not work correctly as it tries to apply structure to pointer without dereferencing.

**Comment by hugsy** [comment]
> Can we call apply_structure_to_address recursively for address that pointer points to and not just remove LP_ part?
> 
> https://github.com/hugsy/gef/blob/master/gef.py#L4730
> 

Interesting because when I test this (WIP) functionality, without `LP_` `ctypes` fails to determine the pointer as a valid pointer to a structure (which is what we want). But correctly does when I mention `LP_`. I realize it's super dirty (you may have noticed the `"hack"` mentioned in the code 😁), but I'm happy to take recommendation/PR on this.

Basically, if I remove the `LP_` part, `ctypes` assume the structure embeds another structure, not a pointer to it.

**Comment by hugsy** [comment]
@xmikasax I misunderstood your point, you were correct so I brought the changes to 669baa1edba7111bca115cbc9226a8512c9db321 and it works as intended:

test file:
```c
#include <stdio.h>
#include <stdlib.h>
#include <stdint.h>

struct A
{
        uint32_t A_a;
        uint64_t A_b;
        uint16_t A_c;
};

struct B
{
        uint16_t B_a;
        struct A* B_b;
};

int main()
{
        struct A* p1 = (struct A*)calloc(1, sizeof(struct A));
        struct B* p2 = (struct B*)calloc(1, sizeof(struct B));
        p1->A_a = 1234;
        p1->A_b = 1234567890;
        p1->A_c = 12345;
        p2->B_a = 1111;
        p2->B_b = p1;
        printf("p1=%p\np2=%p\n", p1, p2);
        __asm__( "int3" );
        return 0;
}
```

produces in `gef`:
```
[...]
p2=0x5555555592c0
[...]
gef ➤  pcustom B 0x5555555592c0
0x5555555592c0+0x0000 B_a :             1111 (c_ushort)
0x5555555592c0+0x0008 B_b :             <A.LP_A object at 0x7f2968187240> (LP_A)
  0x5555555592a0+0x0000 A_a :             1234 (c_uint)
  0x5555555592a0+0x0008 A_b :             0x00000000499602d2 (c_ulong)
  0x5555555592a0+0x0010 A_c :             12345 (c_ushort)
```

Feel free to test it out and let me know (also note that the `pcustom` has slightly changed in syntax to benefit from GDB subcommands). Doc updates are on the way...

Cheers,

**Comment by hugsy** [comment]
This was fixed by 669baa1
